libminifi/include/utils/BaseHTTPClient.h

```diff
@@ -178,6 +178,12 @@ namespace HTTPRequestResponse {
   int seek_callback(void *p, int64_t offset, int);
 }
 
+#undef DELETE  // this is a macro in winnt.h
```

Comment:
I think this should be `#undef`-ed nearest to the source, in the file that includes `winnt.h`?

Reply:
The macro comes from `SSLContextService.h` <- `windows.h` <- `windef.h` <- `minwindef.h` <- `winnt.h`.

We could add the `#undef` in SSLContextService, but that could easily break later.  Or we could replace all 30+ `#include <windows.h>` lines with a wrapper, but that should be a separate PR (and could still break if someone in the future forgets to use the wrapper).

I think the easiest and safest option is to do the `#undef` here.